# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeBuilderCppSupplement.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 parser/html/nsHtml5TreeBuilderCppSupplement.h
--- a/parser/html/nsHtml5TreeBuilderCppSupplement.h	Tue May 14 16:43:27 2024 +0000
+++ b/parser/html/nsHtml5TreeBuilderCppSupplement.h	Tue May 14 16:43:28 2024 +0000
@@ -1722,7 +1722,7 @@
   if (mBuilder) {
     nsIContent* root = nsContentUtils::AttachDeclarativeShadowRoot(
         static_cast<nsIContent*>(aHost), mode, aShadowRootIsClonable,
-        aShadowRootDelegatesFocus);
+        aShadowRootIsSerializable, aShadowRootDelegatesFocus);
     if (!root) {
       nsContentUtils::LogSimpleConsoleError(
           u"Failed to attach Declarative Shadow DOM."_ns, "DOM"_ns,
@@ -1738,9 +1738,9 @@
     return nullptr;
   }
   nsIContentHandle* fragHandle = AllocateContentHandle();
-  opGetShadowRootFromHost operation(aHost, fragHandle, aTemplateNode, mode,
-                                    aShadowRootIsClonable,
-                                    aShadowRootDelegatesFocus);
+  opGetShadowRootFromHost operation(
+      aHost, fragHandle, aTemplateNode, mode, aShadowRootIsClonable,
+      aShadowRootIsSerializable, aShadowRootDelegatesFocus);
   treeOp->Init(mozilla::AsVariant(operation));
   return fragHandle;
 }